packages/terra-application-navigation/src/ApplicationNavigation.jsx

```diff
@@ -0,0 +1,367 @@
+  onSelectExtensionItem,
+  onSelectSettings,
+  onSelectHelp,
+  onSelectLogout,
+  utilityItems,
+  onSelectUtilityItem,
+  notifications,
+  children,
+}) => {
+  const drawerMenuRef = useRef();
+  const contentLayoutRef = useRef();
+  const mainContainerRef = useRef();
+  const utilityButtonPopupAnchorRef = useRef();
+  const drawerMenuIsVisibleRef = useRef(false);
+  const closeMenuCallbackRef = useRef();
+
+  const [drawerMenuIsOpen, setDrawerMenuIsOpen] = useState(false);
+  const [popupMenuIsOpen, setPopupMenuIsOpen] = useState(false);
+
+  // Use dot notation temporarily until hooks + enzyme support for userContext
+  const activeBreakpoint = React.useContext(ActiveBreakpointContext);
+
+  /**
+   * Given a callback function, generateMenuClosingCallback will return a new function
+   * that will ensure that the various menu states are reset before the callback function
+   * is executed.
+   */
+  function generateMenuClosingCallback(wrappedFunction) {
+    return (...args) => {
+      if (!wrappedFunction) {
+        return;
+      }
+
+      closeMenuCallbackRef.current = () => { wrappedFunction(...args); };
+      setDrawerMenuIsOpen(false);
+      setPopupMenuIsOpen(false);
+    };
+  }
+
+  function focusMainContent() {
+    mainContainerRef.current.setAttribute('tabindex', '-1');
+    mainContainerRef.current.focus();
+    mainContainerRef.current.removeAttribute('tabindex');
+  }
+
+  function renderDrawerMenu() {
+    if (!shouldRenderCompactNavigation(activeBreakpoint)) {
+      return null;
+    }
+
+    return (
+      <FocusTrap
+        active={drawerMenuIsOpen}
+        focusTrapOptions={{
+          escapeDeactivates: true,
+          clickOutsideDeactivates: true,
+          returnFocusOnDeactivate: true,
+          onDeactivate: () => setDrawerMenuIsOpen(false),
+        }}
+        className={cx('drawer-menu-focus-trap-container')}
+      >
+        <DrawerMenu
+          titleConfig={titleConfig}
+          userConfig={userConfig}
+          hero={hero}
+          navigationItems={navigationItems}
+          activeNavigationItemKey={activeNavigationItemKey}
+          onSelectNavigationItem={onSelectNavigationItem ? generateMenuClosingCallback(onSelectNavigationItem) : undefined}
+          onSelectSettings={onSelectSettings ? generateMenuClosingCallback(onSelectSettings) : undefined}
+          onSelectHelp={onSelectHelp ? generateMenuClosingCallback(onSelectHelp) : undefined}
+          onSelectLogout={onSelectLogout ? generateMenuClosingCallback(onSelectLogout) : undefined}
+          utilityItems={utilityItems}
+          onSelectUtilityItem={onSelectUtilityItem ? generateMenuClosingCallback(onSelectUtilityItem) : undefined}
+          notifications={notifications}
+        />
+      </FocusTrap>
+    );
+  }
+
+  function renderPopupMenu() {
+    return (
+      <Popup
+        attachmentBehavior="push"
+        contentAttachment="top right"
+        contentHeight="auto"
+        contentWidth="320"
```

Comment:
If we drop this down to `300` it will allow whitespace on either side of the menu on mobile devices with viewport widths of 320. 320px is the common width for older iPhones.

Here is how it renders currently
<img width="413" alt="Screen Shot 2019-06-09 at 6 06 45 AM" src="https://user-images.githubusercontent.com/633148/59158258-fe632900-8a7c-11e9-98e4-9504a3dd1e46.png">

Reply:
Might be a good UX topic. 320px is a standard fixed size within popup, and it was the width used in the zeplin callouts, but it'd be a good question for if we want to adjust popup widths across the board by 20px to account for insets or not.

Reply:
cc/ @neilpfeiffer @jeremyfuksa Any thoughts on shrinking the width on this some?

Reply:
I don't want to block the PR on this but it is something I'm curious about. Don't let this comment hold up the PR.